File: Corrections/java/Neufplate/src/main/java/fr/neufplate/AvatarProvider/RobotHash/RobotHashClient.java
package fr.neufplate.AvatarProvider.RobotHash;

import fr.neufplate.AvatarClientInterface;

public final class RobotHashClient implements AvatarClientInterface {

    private static RobotHashClient instance;

    private static final String apiUrl = "https://robohash.org";
    private static final String fileType = ".png";

    public static RobotHashClient getInstance() {
        if (instance == null) {
            instance = new RobotHashClient();
        }
        return instance;
    }

    public String getRandomAvatarUrl() {
        return apiUrl + "/" + Math.random() + fileType;
    }

    @Override
    public String getAvatarFromHash(String hash) {
        return apiUrl + "/" + hash + fileType;
    }


}

File: Corrections/php/Neufplate/src/AvatarProvider/RobotHash/RobotHashClient.php
<?php

namespace App\AvatarProvider\RobotHash;

use App\AvatarClientInterface;
use App\AvatarProvider\SpriteType;

class RobotHashClient implements AvatarClientInterface
{
    private static ?RobotHashClient $_instance = null;

    public string $spriteType;
    private const apiUrl = "https://robohash.org";
    private const fileType = ".png";

    public function __construct()
    {
        $this->spriteType = SpriteType::DEFAULT->value;
    }

    public static function getInstance(): AvatarClientInterface {

        if(is_null(self::$_instance)) {
            self::$_instance = new RobotHashClient();
        }

        return self::$_instance;
    }

    // Exercice 2
    public function getRandomAvatarUrl(): string
    {
        return self::apiUrl . "/" . uniqid() . self::fileType . PHP_EOL;
    }

    public function getAvatarFromHash(string $hash): string
    {
        return self::apiUrl . "/" . $hash . self::fileType . PHP_EOL;
    }
}